### PR TITLE
[TOPI] Add generic batch norm

### PR DESCRIPTION
**Pull request by michalpiszczek**
- Add a TOPI implementation of batch norm so it can be constant folded
- Add associated tests

### CODE REVIEW

**Comment by michalpiszczek** [comment]
@mbrookhart PTAL

**Comment by michalpiszczek** [comment]
@AndrewZhaoLuo @jwfromm PTAL

File: tests/python/relay/test_op_level1.py [hunk 1]
```diff
@@ -427,6 +428,52 @@ def test_batch_norm():
...
+        relay.const(beta_np),
+        relay.const(moving_mean_np),
+        relay.const(moving_var_np),
+        axis=axis,
+    ).astuple()
+    func_const = relay.Function([], out_const)
+
+    # Build the module with constants to have FoldConstant transform batch_norm.
+    mod_const = tvm.IRModule.from_expr(func_const)
+    lib_const = relay.build(mod_const, tvm.target.create("llvm"))
```
Comment (mbrookhart) [inline tests/python/relay/test_op_level1.py:462]: When compiling, SimplifyInference runs before fold constant. Can you manually FoldConstant here instead, and compare that output to the VM run below?

https://github.com/apache/tvm/blob/main/src/relay/backend/utils.cc#L196-L226

**Review by mbrookhart — CHANGES_REQUESTED** [review]
I think there's a problem with the relay test, but otherwise it looks fantastic. Thank you!

File: tests/python/relay/test_op_level1.py (hunk 1 above)
Comment (michalpiszczek) [reply tests/python/relay/test_op_level1.py]: Good catch, fixed!

**Review by mbrookhart — APPROVED** [review]
LGTM!

**Pull request merged**